java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java

```diff
@@ -1905,10 +1953,34 @@ protected int diffCase(JCCase oldT, JCCase newT, int[] bounds) {
         if (localPointer < endPos(oldT)) {
             copyTo(localPointer, localPointer = endPos(oldT));
         }
+        if (reindentStatements) {
+            printer.reindentRegions = oldReindentRegions;
+            printer.reindentRegions.add(new int[] {reindentStart, localPointer});
+        } else {
+            oldReindentRegions.addAll(printer.reindentRegions);
+            printer.reindentRegions = oldReindentRegions;
+        }
         printer.undent(old);
         return localPointer;
     }
 
+    public static List<JCExpression> getCasePatterns(JCCase cs) {
```

Comment:
Shouldn't these go to `TreeShims` ?

Reply:
It could, but method in TreeShims may fail (and I think that's a good thing), while these method will not fail.